packages/instantsearch-ui-components/src/components/recommend-shared/ListView.tsx

```diff
@@ -0,0 +1,48 @@
+/** @jsx createElement */
+
+import type {
+  RecommendClassNames,
+  RecommendTranslations,
+  RecordWithObjectID,
+  Renderer,
+  RecommendViewProps,
+} from '../../types';
+
+export function createListViewComponent({ createElement, Fragment }: Renderer) {
+  return function ListView<TItem extends RecordWithObjectID>(
+    userProps: RecommendViewProps<
+      TItem,
+      RecommendTranslations,
+      Partial<RecommendClassNames>
+    >
+  ) {
+    const {
+      classNames = {},
+      itemComponent: ItemComponent,
+      items,
+      sendEvent,
+    } = userProps;
+
+    return (
+      <div className={classNames.container}>
+        <ol className={classNames.list}>
+          {items.map((item) => (
+            <ItemComponent
+              key={item.objectID}
+              className={classNames.item}
+              createElement={createElement}
+              Fragment={Fragment}
```

Comment:
There should be no need to pass these in InstantSearch.js as we only encourage to use `htm`